vllm/engine/async_llm_engine.py

```diff
@@ -211,12 +211,17 @@ async def step_async(self) -> List[RequestOutput]:
 
         if not scheduler_outputs.is_empty():
             # Execute the model.
+            execute_model_data = ExecuteModelData(
```

Comment:
QQ: why don't we just include seq_group_metadata_list to this data structure too and call it ExecuteModelRequest?

Reply:
Good question. I just feel this might be a clear separation between real input data and other metadata, as I found other components such as model runner only takes seq group. However, I don't have a strong preference so I can also include seq group if you prefer.

Reply:
And for naming, if we include seq group, ExexuteDataRequest seems ok to me; otherwise ExecuteMetadata seems more aligned

Reply:
I'd vote for just a single request input (since it is like a request between scheduler <> executor). But @cadedaniel plz chime in if you disagree

Reply:
yeah IMO the lines are blurry between metadata and data because seq_group_metadata_list also has metadata about the request. +1 for single request input but will approve PR either way

Reply:
Sure let's make it a single pack and name it ExecuteDataRequest :)